protocols/bgp/server/bmp_server.go

```diff
@@ -52,15 +53,116 @@ func NewServer(keepalivePeriod time.Duration) *BMPServer {
 	return b
 }
 
+// Listen starts a listener for routers to start the BMP connection
+// the listener needs to be closed by calling Close() on the BMPServer
+func (b *BMPServer) Listen(addr string) error {
+	tcp, err := net.ResolveTCPAddr("tcp", addr)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"component": "bmp_server",
+			"address":   addr,
+		}).Info("Unable to resolve address")
+		return err
+	}
+	l, err := net.ListenTCP("tcp", tcp)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"component": "bmp_server",
+			"address":   addr,
+		}).Infof("Unable to listen on %s", tcp.String())
+		return err
+	}
+	b.listener = l
+
+	for {
+		c, err := l.Accept()
+		if err != nil {
+			log.WithError(err).WithFields(log.Fields{
+				"component": "bmp_server",
+				"address":   addr,
+			}).Infof("Unable to accept on %s", tcp.String())
+			return err
+		}
+		go b.handleConnection(c)
+	}
+}
+
+func (b *BMPServer) handleConnection(c net.Conn) {
+	remoteIP := c.RemoteAddr().(*net.TCPAddr).IP
+
+	var r *Router
+	for _, router := range b.getRouters() {
```

Comment:
Let's use the GetRouter() method here. Then just check for passive flag.

Reply:
Thanks, changed it.